### DISCUSSION TITLE
How to stream the simulation to other softwares

### DISCUSSION THREAD

**Discussion by carlosedubarreto** [NVIDIA-Omniverse/PhysX#24] [Q&A] [answered]
I would like to use Physx to run the simulation and send the result in realtime to other applications, how can it be done?

I would like at least a direction to look, because I cant see how to do that, and I'm also a beginner on C++, so its all new to me.

**Comment by amoravanszky** [comment, marked as answer]
There is no feature in PhysX that will just do it for you, you will need to get the updated state from PhysX using the same functions the snippets (sample programs) use to retrieve updated transforms and send them to the renderer; except you will need to send these transforms over a socket or other interface to your other application, for which you will typically need to write some sort of plugin to receive the data.  You will also need to worry about object life management on both sides of the connection.

If you are a beginner you might be biting off more here than you can chew, honestly.  I would start with an easier project to learn C++, because if you do this, you will simultaneously be learning, C++, PhysX, networking, and how to extend your target application, all at the same time.

**Reply by carlosedubarreto** [reply]
Thanks a lot for the answer.
I'm hours researching about it, and I was getting that exactly conclusion ( I didnt think about the object life management )

And a tip for others that might read this.
Take a good read at the Physx documentation, I spent hours searching for things that was already there extremelly visible (like the Api Basics part https://nvidia-omniverse.github.io/PhysX/physx/5.1.0/docs/API.html )


I'm learning more by deeply studying  the snippets, they are just amazing.

**Comment by Alan-FGR** [comment]
I'm not sure you came across [this series of articles](https://gafferongames.com/post/introduction_to_networked_physics/), but it's a must-read, especially since you said you're spending time on research it might save you a lot of it.
I learned the hard way that for instance extrapolation doesn't really work as well as one might expect, it might be useful however to extrapolate a reference point for other calculations. Anyway, hope that helps.

**Reply by carlosedubarreto** [reply]
Thanks. I didn't get to this article. I'm more on the beginning part, the network parte I'll probably take much longer to start.
The article looks great.
